teamcity-rest-client-api/src/main/kotlin/org/jetbrains/teamcity/rest/api.kt

```diff
@@ -648,7 +648,7 @@ enum class TestStatus {
     UNKNOWN
 }
 
-interface TestOccurrence {
+interface TestRun {
```

Comment:
This rename breaks source and binary compatibility, and we tend to keep compatibility while doing changes (see [CONTRIBUTING.md](https://github.com/JetBrains/teamcity-rest-client/blob/master/CONTRIBUTING.md) for details).

In order to proceed with the rename gracefully, you can keep `TestOccurrence` as deprecated, extend the new interface `TestRun` from it, and use `TestRun` in the new code.

It's a question what to do with `Build::tests` method. We can either deprecate it, introduce a new method which returns `Sequence<TestRun>` and delegate the old method to the new. But since we cannot have two methods which signatures differ in return type only, we'll need to also invent a new name for the new method (`testRuns`?). The other option is to change the return type to `Sequence<TestRun>` as you did. This change is binary compatible because generic parameters are erased from classfiles, and even source compatible for clients written in Kotlin (because `Sequence` is covariant by its generic parameter) and we suppose that clients only calls API methods, don't implement them. However it is source incompatible for clients written in Java. So I'm not sure how to solve this properly. Does `testRuns` name look good for you? If so I suggest to choose the first variant.